### PR TITLE
feat(project): Add gather data

### PR DESCRIPTION
**Pull request by jeremyvergnas**
feat(project): Add gather data

Useful when we need to export all the data of the datatable for instance.

https://sweebr.atlassian.net/browse/CM-66

### CODE REVIEW

File: src/datatable.js [hunk 1]
```diff
@@ -91,6 +91,14 @@ export class DataTable {
       });
   }
 
+  gatherData(criteria = {}) {
+    return this.repository.find(criteria || {}, true).then(result => {
```
Comment (doktordirk) [inline src/datatable.js]: either criteria = {} or criteria || {}

File: src/datatable.js [hunk 2]
```diff
@@ -91,6 +91,14 @@ export class DataTable {
...
+      return result;
+    }).catch(error => {
```
Comment (doktordirk) [inline src/datatable.js]: the then result part isn't doing anything

File: src/datatable.js [hunk 3]
```diff
@@ -91,6 +91,14 @@ export class DataTable {
...
+      this.triggerEvent('exception', {on: 'load', error: error});
+    });
```
Comment (doktordirk) [inline src/datatable.js:97]: tricky. re.throw the error or not. i see you decided against @RWOverdijk thoughts. i guess when the event is used, re-throwing would just annoy

File: src/bootstrap/datatable.html [hunk 4]
```diff
@@ -52,7 +52,7 @@
 
         <!-- Columns -->
         <td repeat.for="columnLabel of columnLabels">
-          <a click.delegate="selected(row)">${displayValue(row, columnLabel.column) | convertManager: columnLabel.converter}</a>
+          <a click.delegate="selected(row)" innerhtml.bind="displayValue(row, columnLabel.column) | convertManager: columnLabel.converter"></a>
```
Comment (doktordirk) [inline src/bootstrap/datatable.html:55]: what's the motivation? adding sanitizeHTML converter would have been one

**Review by doktordirk — CHANGES_REQUESTED** [review]

File: src/bootstrap/datatable.html (hunk 4 above)
Comment (jeremyvergnas) [reply src/bootstrap/datatable.html:55]: We want to be able to use HTML tags in our column. In our case we need it to show barcodes.

File: src/datatable.js (hunk 1 above)
Comment (jeremyvergnas) [reply src/datatable.js]: Indeed

File: src/datatable.js (hunk 3 above)
Comment (jeremyvergnas) [reply src/datatable.js:97]: It is using the same pattern we use all over the datatable plugin.

**Comment by VMBindraban** [comment]
Update docs with new feature.

File: src/bootstrap/datatable.html (hunk 4 above)
Comment (doktordirk) [reply src/bootstrap/datatable.html:55]: i see. a separate commit (message) would have been nice though. 

i wounder if not `| sanitize` could/should be added. thoughts?

**Review by VMBindraban — CHANGES_REQUESTED** [review]
Update docs with new feature.

**Review by doktordirk — APPROVED** [review]

**Comment by VMBindraban** [comment]
Fixes https://github.com/SpoonX/aurelia-datatable/issues/59

File: src/bootstrap/datatable.html (hunk 4 above)
Comment (jeremyvergnas) [reply src/bootstrap/datatable.html:55]: You probably mean the value converter `sanitizeHTML`. For some reason, it would break the datatable...

**Pull request merged**